### ISSUE TITLE
Socket: urlParts parsing defect when sockPath is set

### ISSUE THREAD

**Issue by dennisreimann** [webpack/webpack-dev-server#2357]
- Operating System: macOS
- Node Version: 10.15
- NPM Version: 6.4.1
- webpack Version: 4.41.3
- webpack-dev-server Version: 3.9.0
- Browser: Firefox Dev Edition 72.0b7

- [x] This is a **bug**
- [ ] This is a **modification** request

### Code

```js
// webpack.config.js
module.exports = {
  devServer: {
    transportMode: 'ws', // also happens when transport mode is not set
    sockPath: '/'
  }
}
```

### For Bugs; How can we reproduce the behavior?

Either the `resourceQuery` that gets fed to the `createSocketUrl` function is wrong formatted or the `urlParts` parsing which handles the `resourceQuery` does not work as expected.

This is what it looks like without `sockPath` configuration (correct):

<img width="432" alt="resourcequery-01" src="https://user-images.githubusercontent.com/886/71032490-bf8dea00-2115-11ea-870c-4337c3bc6923.png">

And it does not work when the `sockPath` is set:

<img width="669" alt="resourcequery-02" src="https://user-images.githubusercontent.com/886/71032545-dc2a2200-2115-11ea-96dc-231f3e4f77d1.png">

**Comment by dennisreimann** [comment]
Also: When setting `transportMode: 'ws'` I still get requests to `/sockjs-node` – actually that's why I tried to set the `sockPath` in the first place. So I guess there is something not quite correct as well, but I'm unsure how all of this relates.
I can provide more details if necessary.

**Comment by evilebottnawi** [comment]
Please create reproducible test repo

**Comment by evilebottnawi** [comment]
hm, can you try master branch?

**Comment by dennisreimann** [comment]
I can confirm that the parsing part is fixed by [this commit](a599f99adaac012dcf97b67c796d7890b2096bee).

Also checked that [these changes](e188542d888dbb55be64c9da2f747343b73c319f) work as intended, so that a custom `sockPath` can be used.

All in all I can use `transportMode: 'ws'` without specifying the `sockPath`, which is great :)

Thanks for the quick response!
